pennylane/qchem/openfermion_obs.py

```diff
@@ -649,6 +658,196 @@ def dipole_of(
+            corresponding to the qubit number equal to its index.
+            For type dict, only int-keyed dict (for qubit-to-wire conversion) is accepted for
+            partial mapping. If None, will use identity map.
+        args (array[array[float]]): Initial values of the differentiable parameters
+        cutoff (float): Cutoff value for including the matrix elements
+            :math:`\langle \alpha \vert \hat{{\bf r}} \vert \beta \rangle`. The matrix elements
+            with absolute value less than ``cutoff`` are neglected.
+
+    Returns:
+        list[pennylane.Hamiltonian]: The qubit observables corresponding to the components
+        :math:`\hat{D}_x`, :math:`\hat{D}_y` and :math:`\hat{D}_z` of the dipole operator.
+
+    **Example**
+
+    >>> symbols = ["H", "H", "H"]
+    >>> coordinates = np.array([[0.028, 0.054, 0.0], [0.986, 1.610, 0.0], [1.855, 0.002, 0.0]])
+    >>> mol = qml.qchem.Molecule(symbols, coordinates, charge=1)
+    >>> dipole_obs = qml.qchem.molecular_dipole(mol, method="openfermion")
+    >>> dipole_obs[0] # x-component of D
+    (
+        0.4781123173263876 * Z(0)
+      + 0.4781123173263876 * Z(1)
+      + -0.3913638489489803 * (Y(0) @ Z(1) @ Y(2))
+      + -0.3913638489489803 * (X(0) @ Z(1) @ X(2))
+      + -0.3913638489489803 * (Y(1) @ Z(2) @ Y(3))
+      + -0.3913638489489803 * (X(1) @ Z(2) @ X(3))
+      + 0.2661114704527088 * (Y(0) @ Z(1) @ Z(2) @ Z(3) @ Y(4))
+      + 0.2661114704527088 * (X(0) @ Z(1) @ Z(2) @ Z(3) @ X(4))
+      + 0.2661114704527088 * (Y(1) @ Z(2) @ Z(3) @ Z(4) @ Y(5))
+      + 0.2661114704527088 * (X(1) @ Z(2) @ Z(3) @ Z(4) @ X(5))
+      + 0.7144779061810713 * Z(2)
+      + 0.7144779061810713 * Z(3)
+      + -0.11734958781031017 * (Y(2) @ Z(3) @ Y(4))
+      + -0.11734958781031017 * (X(2) @ Z(3) @ X(4))
+      + -0.11734958781031017 * (Y(3) @ Z(4) @ Y(5))
+      + -0.11734958781031017 * (X(3) @ Z(4) @ X(5))
+      + 0.24190977644645698 * Z(4)
+      + 0.24190977644645698 * Z(5)
+    )
+    """
+
+    if method not in ["dhf", "openfermion"]:
+        raise ValueError("Only 'dhf', and 'openfermion' backends are supported.")
+
+    if mapping.strip().lower() not in ["jordan_wigner", "parity", "bravyi_kitaev"]:
+        raise ValueError(
+            f"'{mapping}' is not supported."
+            f"Please set the mapping to 'jordan_wigner', 'parity' or 'bravyi_kitaev'."
+        )
+
+    symbols = molecule.symbols
+    coordinates = molecule.coordinates
+
+    if qml.math.shape(coordinates)[0] == len(symbols) * 3:
+        geometry_dhf = qml.numpy.array(coordinates.reshape(len(symbols), 3))
+        geometry_hf = coordinates
+    elif len(coordinates) == len(symbols):
+        geometry_dhf = qml.numpy.array(coordinates)
+        geometry_hf = coordinates.flatten()
+
+    if molecule.mult != 1:
+        raise ValueError(
+            "Open-shell systems are not supported. Change the charge or spin multiplicity of the molecule."
+        )
+
+    wires_map = None
```

Comment:
Why do we define this above? 🤔 More importantly, `wires_map` is only being used for `qml.map_wires`, so why not use it directly there?

Reply:
Good point, thanks for noticing this.